src/podofo/main/PdfPage.h

```diff
@@ -37,6 +37,9 @@ struct PODOFO_API PdfTextExtractParams final
 {
     nullable<Rect> ClipRect;
     PdfTextExtractFlags Flags = PdfTextExtractFlags::None;
+
+    std::function<bool(int read_cnt, void *d)> AbortCheck = nullptr;
+    void *AbortCheckData = nullptr;
```

Comment:
That `AbortCheckData` stuff is pretty outdated in terms of C++ style. It's more idiomatic to have a callback `std::function<bool(int)>` and capture any additional data needed in a lambda.